src/lib/pubkey/curve448/ed448/ed448_internal.cpp

```diff
@@ -0,0 +1,317 @@
+
+/*
+ * Ed448 Internals
+ * (C) 2024 Jack Lloyd
+ *     2024 Fabian Albert - Rohde & Schwarz Cybersecurity
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ */
+#include <botan/internal/ed448_internal.h>
+
+#include <botan/exceptn.h>
+#include <botan/types.h>
+#include <botan/internal/ct_utils.h>
+#include <botan/internal/loadstor.h>
+#include <botan/internal/mp_core.h>
+#include <botan/internal/shake_xof.h>
+#include <botan/internal/stl_util.h>
+
+namespace Botan {
+namespace {
+
+constexpr uint64_t MINUS_D = 39081;
+
+secure_vector<uint8_t> dom4(uint8_t x, std::span<const uint8_t> y) {
```

Comment:
`secure_vector` seems odd here since this is basically a constant (and certainly public)